src/agentscope/model/_zhipu_model.py

```diff
@@ -0,0 +1,416 @@
+            ) from e
+
+        super().__init__(model_name, stream)
+
+        self.client = ZhipuAiClient(
+            api_key=api_key,
+            base_url=base_url,
+            **kwargs,
+        )
+        self.thinking = thinking
+        self.generate_kwargs = generate_kwargs or {}
+
+    @trace_llm
+    async def __call__(
+        self,
+        messages: list[dict[str, Any]],
+        tools: list[dict] | None = None,
+        tool_choice: Literal["auto", "none", "any", "required"]
+        | str
+        | None = None,
+        structured_model: Type[BaseModel] | None = None,
+        **kwargs: Any,
+    ) -> ChatResponse | AsyncGenerator[ChatResponse, None]:
+        """Get the response from Zhipu AI chat completions API by the given
+        arguments.
+
+        Args:
+            messages (`list[dict]`):
+                A list of dictionaries, where `role` and `content` fields are
+                required, and `name` field is optional.
+            tools (`list[dict]`, default `None`):
+                The tools JSON schemas that the model can use.
+            tool_choice (`Literal["auto", "none", "any", "required"] | str \
+                | None`, default `None`):
+                Controls which (if any) tool is called by the model.
+                Can be "auto", "none", or specific tool name.
+            structured_model (`Type[BaseModel] | None`, default `None`):
+                A Pydantic BaseModel class that defines the expected structure
+                for the model's output.
+            **kwargs (`Any`):
+                The keyword arguments for Zhipu AI chat completions API,
+                e.g. `temperature`, `max_tokens`, `top_p`, etc. Please refer to
+                the Zhipu AI API documentation for more details.
+
+        Returns:
+            `ChatResponse | AsyncGenerator[ChatResponse, None]`:
+                The response from the Zhipu AI chat completions API.
+        """
+
+        # Validate messages
+        if not isinstance(messages, list):
+            raise ValueError(
+                "Zhipu AI `messages` field expected type `list`, "
+                f"got `{type(messages)}` instead.",
+            )
+        # Allow messages with only tool_calls and no content.
+        # Only validate that messages contain the role field.
+        if not all("role" in msg for msg in messages):
+            raise ValueError(
+                "Each message in the 'messages' list must contain a 'role' "
+                "key for Zhipu AI API.",
+            )
+
+        kwargs = {
+            "model": self.model_name,
+            "messages": messages,
+            "stream": self.stream,
+            **self.generate_kwargs,
+            **kwargs,
+        }
+
+        # Add thinking parameter if supported and not already in kwargs
+        if self.thinking and "thinking" not in kwargs:
+            # Note: Zhipu AI may support thinking parameters in future versions
+            logger.warning(
```

Comment:
Please remove this warning here, since  Zhipu AI does support the `thinking` parameter.